web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php

```diff
@@ -136,20 +130,30 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
     $form['reservations']['reservation_sms_notifications_enabled'] = [
       '#type' => 'checkbox',
       '#title' => $this->t('Enable SMS notifications for reservations', [], ['context' => 'Library Agency Configuration']),
-      '#default_value' => $config->get('reservation_sms_notifications_enabled') ?? self::RESERVATION_SMS_NOTIFICATIONS_ENABLED,
+      '#default_value' => $config->get('reservation_sms_notifications_enabled') ?? GeneralSettings::RESERVATION_SMS_NOTIFICATIONS_ENABLED,
       '#description' => $this->t('If checked, SMS notifications for patrons are enabled.', [], ['context' => 'Library Agency Configuration']),
     ];
 
     $form['reservations']['interest_periods_config'] = [
       '#type' => 'textarea',
+      '#required' => TRUE,
       '#title' => $this->t('Interest periods for reservation', [], ['context' => 'Library Agency Configuration']),
-      '#default_value' => $config->get('interest_periods_config') ?? self::INTEREST_PERIODS_CONFIG,
+      '#default_value' => $config->get('interest_periods_config') ?? GeneralSettings::INTEREST_PERIODS_CONFIG,
+      '#description' => $this->t('Set the interest periods. The format should be [days]-[label]. New periods will first be available in "Default interest period for reservation" when the form has been saved.', [], ['context' => 'Library Agency Configuration']),
+    ];
+
+    $form['reservations']['default_interest_period_config'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Default interest period for reservation', [], ['context' => 'Library Agency Configuration']),
+      '#options' => $this->generalSettings->getInterestPeriodsAsArray(),
+      '#default_value' => $config->get('default_interest_period_config') ?? GeneralSettings::DEFAULT_INTEREST_PERIOD_CONFIG,
+      '#description' => $this->t('Set the default interest period for reservations.', [], ['context' => 'Library Agency Configuration']),
```

Comment:
I don't think it is the best solution with one field being depending on that you have submitted another. It is a confusing user experience, I think.
But I underrstand why you did it, and it is a viable MVP for now, I think. If the customer wants changes here they must telle us. I think we should telle them that we chose to go this way in our next demo

Reply:
I agree that this is not the best user experience. I tried to do some ajax magic, but had some problems with it. Decided it was not worth the time as I have already used enough.

Reply:
👍